### PR TITLE
Add text, scheme_text & attack_text that are missing

### PR DESCRIPTION
**Pull request by Jensen1989**

### CODE REVIEW

File: pack/mts_encounter.json [hunk 1]
```diff
@@ -673,6 +690,8 @@
 		"quantity": 1,
 		"set_code": "tower_defense",
 		"set_position": 9,
+		"text": "The unique rule does not apply to Avenger's Tower. \n <b>Forced Response:</b> After damage is placed here, if there is at least 9[per_player] damage here, remove all of it. Then flip Avenger's Tower over.",
```
Comment (lostie) [inline pack/mts_encounter.json:693]: I believe when cards have an `A` and `B` side (which this one have as you can see my the card ID in the bottom left), the `A` side should have the `back_text` and the `B` side should have the `text`.

```suggestion
		"back_text": "The unique rule does not apply to Avengers Tower.\n<b>Forced Response:</b> After damage is placed here, if there is at least 9[per_player] damage here, remove all of it. Then flip Avenger's Tower over.",
```

File: pack/mts_encounter.json [hunk 2]
```diff
@@ -673,6 +690,8 @@
...
+		"back_text": "<b> When Revealed: </b> Discard each other Avengers Tower from play. \n <b> Forced Response: </b> After damage is placed here ,if there is at least 9[per player] damage here, ther players lose the game.",
```
Comment (lostie) [inline pack/mts_encounter.json:694]:
```suggestion
		"text": "<b>When Revealed:</b> Discard each other Avengers Tower from play.\n<b> Forced Response:</b> After damage is placed here, if there is at least 9[per player] damage here, the players lose the game.",
```

File: pack/mts_encounter.json [hunk 3]
```diff
@@ -686,6 +705,7 @@
 		"quantity": 1,
 		"set_code": "tower_defense",
 		"set_position": 10,
+		"text": "Permanent. \n The villain who matches the attached scheme is the active villain. \n <b> Forced Response: </b> After the player phase ends, attach this card to the other main scheme.",
```
Comment (lostie) [inline pack/mts_encounter.json:708]:
```suggestion
		"text": "Permanent.\n The villain who matches the attached scheme is the active villain.\n<b>Forced Response:</b> After the player phase ends, attach this card to the other main scheme.",
```

**Review by lostie — COMMENTED** [review]
I was only able to review just the beginning of the PR (it's quite a lot of card text to review :P ) there's probably some patterns I've mentioned that apply to some other card data I was not able to review.

Thanks for taking the time to fill all this information.

File: pack/mts_encounter.json [hunk 4]
```diff
@@ -580,6 +590,7 @@
 		"set_code": "tower_defense",
 		"set_position": 4,
 		"stage": 1,
+		"text": "[star]<b>Forced Interrupt:</b> After Corvus Glaive makes an undefended attack, discard the top card of the encounter deck → deal 1 damange to Avenger's Tower for each boost icon [[boost]] on that card. \n <b> Corvus Glaive cannot be defeated while Proxima Midnight has any hit points remaining </b>",
```
[the remark this answers is not in the document]
Comment (lostie) [reply pack/mts_encounter.json:593]: These text changes should apply to all versions of Corvus Glaive.

File: pack/mts_encounter.json [hunk 5]
```diff
@@ -523,6 +530,7 @@
 		"set_code": "tower_defense",
 		"set_position": 1,
 		"stage": 1,
+		"text": "[star]<b>Forced Interrupt:</b> When Proxima Midnight attacks you, choose to either deal 1 damange to Avenger's Tower, or Proxima Midnight gets +2 ATK for this attack. \n <b> Proxima Midnight cannot be defeated while Corvus Glaive has any hit points remaining </b>",
```
[the remark this answers is not in the document]
Comment (lostie) [reply pack/mts_encounter.json:533]: These text changes should apply to all versions of Proxima Midnight.

File: pack/mts_encounter.json [hunk 6]
```diff
@@ -473,6 +477,7 @@
 		"scheme": 1,
 		"set_code": "armies_of_titan",
 		"set_position": 1,
+		"text": "Guard.\n<b>When Defeated:</b> Give the villain a tough status card.","boost_text": "[star]<b>Boost:</b> Give the villain a tough status",
```
[the remark this answers is not in the document]
Comment (lostie) [reply pack/mts_encounter.json:480]: There's more examples of card data with the `text` and `boost_text` keys not defined alphabetically.

**Comment by Jensen1989** [comment]
> I was only able to review just the beginning of the PR (it's quite a lot of card text to review :P ) there's probably some patterns I've mentioned that apply to some other card data I was not able to review.
> 
> Thanks for taking the time to fill all this information.

Thanks for the feedback, I am new to this json/github thing, so I'm still getting used to correct formatting.
I will fix up the mistakes you mentioned for the rest of the uploads.
Thanks again for the help :)

**Comment by Jensen1989** [comment]
> I was only able to review just the beginning of the PR (it's quite a lot of card text to review :P ) there's probably some patterns I've mentioned that apply to some other card data I was not able to review.
> 
> Thanks for taking the time to fill all this information.

Also if you could clarify the setup for the main schemes?
Since the main schemes are two sided cards, I did side A as 'scheme_text' and side B as back_text, but I think you want side B to be 'text'?

**Comment by lostie** [comment]
> > I was only able to review just the beginning of the PR (it's quite a lot of card text to review :P ) there's probably some patterns I've mentioned that apply to some other card data I was not able to review.
> > Thanks for taking the time to fill all this information.
> 
> Also if you could clarify the setup for the main schemes? Since the main schemes are two sided cards, I did side A as 'scheme_text' and side B as back_text, but I think you want side B to be 'text'?

I'm just following the formatting done for the existing card pool, as an example from the Core Set, in [The Break-In!](https://marvelcdb.com/card/01097) you can check the JSON definition for it and you will find that the 1A side is defined with the `back_text` and 1B side is defined with `text`.

File: pack/mts_encounter.json [hunk 7]
```diff
@@ -429,6 +430,7 @@
 		"scheme": 2,
 		"set_code": "black_order",
 		"set_position": 2,
+		"text": "Quickstrike.\n<b>Forced Response:</b> After supergiant attacks and damages a character, that character is stunned.",
```
[the remark this answers is not in the document]
Comment (lostie) [reply pack/mts_encounter.json:433]: The name of the character needs to be capitalised, it should be `Supergiant`.

File: pack/mts_encounter.json [hunk 8]
```diff
@@ -458,10 +461,12 @@
 		"quantity": 1,
 		"set_code": "black_order",
 		"set_position": 4,
+		"text": "<b>When Revealed:</b> Each minion engaged with a player activates against that player. Each player who is not engaged with a minion searches the encounter deck and discard pile for a [[Black Order]] minion and puts it into play engaged with them. Shuffle the encounter deck.",
 		"type_code": "treachery"
 	},
 	{
 		"attack": 2,
+		"boost_text": "Give the villain a tough status",
```
Comment (lostie) [inline pack/mts_encounter.json:469]:
```suggestion
		"boost_text": "Give the villain a tough status card.",
```

File: pack/mts_encounter.json [hunk 9]
```diff
@@ -523,6 +532,7 @@
 		"set_code": "tower_defense",
 		"set_position": 1,
 		"stage": 1,
+		"text": "<b>Proxima Midnight cannot be defeated while Corvus Glaive has any hit points remaining.</b>",
```
Comment (lostie) [inline pack/mts_encounter.json:535]: I think you're missing the `attack_text` on this card:

```json
"attack_text": "<b>Forced Interrupt:</b> When Proxima Midnight attacks you, choose to either deal 1 damage to Avenger's Tower, or Proxima Midnight gets +2 ATK for this attack."
```

File: pack/mts_encounter.json [hunk 10]
```diff
@@ -673,6 +690,7 @@
 		"quantity": 1,
 		"set_code": "tower_defense",
 		"set_position": 9,
+		"text": "<b>When Revealed:</b> Discard each other Avengers Tower from play.\n<b> Forced Response:</b> After damage is placed here, if there is at least 9[per player] damage here, the players lose the game.",
```
Comment (lostie) [inline pack/mts_encounter.json:693]:
```suggestion
		"text": "<b>When Revealed:</b> Discard each other Avengers Tower from play.\n<b>Forced Response:</b> After damage is placed here, if there is at least 9[per player] damage here, the players lose the game.",
```

File: pack/mts_encounter.json [hunk 11]
```diff
@@ -702,6 +721,7 @@
 		"scheme": 1,
 		"set_code": "tower_defense",
 		"set_position": 11,
+		"text": "<b> Forced Response: </b> After Black Order Besieger engages you, chose to either deal 1 damange to Avengers Tower or deal 2 damange to your identity.",
```
Comment (lostie) [inline pack/mts_encounter.json:724]:
```suggestion
		"text": "<b>Forced Response:</b> After Black Order Besieger engages you, choose to either deal 1 damage to Avengers Tower or deal 2 damage to your identity.",
```

File: pack/mts_encounter.json [hunk 12]
```diff
@@ -717,6 +737,7 @@
 		"quantity": 1,
 		"set_code": "tower_defense",
 		"set_position": 15,
+		"text": "Attach to Proxima Midnght. \n [star]Proxima Midnight's attach gain overkill and piercing. \n <b> Hero Action: </b> Take 1 damage and spend [energy] [mental] resources → discard this card.",
```
Comment (lostie) [inline pack/mts_encounter.json:740]: The ⭐  text can be defined as:

```json
"attack": 1,
"attack_text": "Proxima Midnight's attacks gain overkill and piercing."
```

```suggestion
		"text": "Attach to Proxima Midnght.\n<b>Hero Action:</b> Take 1 damage and spend [energy] [mental] resources → discard this card.",
```

File: pack/mts_encounter.json [hunk 13]
```diff
@@ -732,6 +753,7 @@
 		"quantity": 1,
 		"set_code": "tower_defense",
 		"set_position": 16,
+		"text": "Attacha to Corvus Glaive. \n Corvus Glaive gains retaliate 1. \n <b> Hero action: <\b> Take 1 damage and spend [energy] [physical] resources → discard this card.",
```
Comment (lostie) [inline pack/mts_encounter.json:756]:
```suggestion
		"text": "Attach to Corvus Glaive.\nCorvus Glaive gains retaliate 1.\n<b>Hero Action:<\b> Take 1 damage and spend [energy] [physical] resources → discard this card.",
```

File: pack/mts_encounter.json [hunk 14]
```diff
@@ -746,10 +768,12 @@
 		"quantity": 2,
 		"set_code": "tower_defense",
 		"set_position": 17,
+		"text": "Attach to the villain who is not the active villain. \n [star] <b> Forced Interrupt: </b> When attached villain attacks, the attacks gains ranged. If that attack defeats an ally, deal 3 damange to Avenger's Tower. At the end of that attach, discard Direct Assault",
```
Comment (lostie) [inline pack/mts_encounter.json:771]:
```json
"attack": 2,
"attack_text": "<b>Forced Interrupt:</b> When attached villain attacks, the attacks gains ranged. If that attack defeats an ally, deal 3 damange to Avengers Tower. At the end of that attack, discard Direct Assault."
```

```suggestion
		"text": "Attach to the villain who is not the active villain.",
```

File: pack/mts_encounter.json [hunk 15]
```diff
@@ -746,10 +768,12 @@
...
 		"traits": "Tactic.",
 		"type_code": "attachment"
 	},
 	{
+		"boost_text": "Add the other villain's SCH and ATK to this villain's SCH and ATK for this activation",
```
Comment (lostie) [inline pack/mts_encounter.json:776]:
```suggestion
		"boost_text": "Add the other villain's SCH and ATK to this villain's SCH and ATK for this activation.",
```

File: pack/mts_encounter.json [hunk 16]
```diff
@@ -759,9 +783,11 @@
 		"quantity": 2,
 		"set_code": "tower_defense",
 		"set_position": 19,
+		"text": "<b> When Revealed: </b> Proxima Midnight activates against you.",
```
Comment (lostie) [inline pack/mts_encounter.json:786]:
```suggestion
		"text": "<b>When Revealed:</b> Proxima Midnight activates against you.",
```

File: pack/mts_encounter.json [hunk 17]
```diff
@@ -759,9 +783,11 @@
...
 		"type_code": "treachery"
 	},
 	{
+		"boost_text": "Add the other villain's SCH and ATK to this villain's SCH and ATK for this activation",
```
Comment (lostie) [inline pack/mts_encounter.json:790]:
```suggestion
		"boost_text": "Add the other villain's SCH and ATK to this villain's SCH and ATK for this activation.",
```

File: pack/mts_encounter.json [hunk 18]
```diff
@@ -771,9 +797,11 @@
 		"quantity": 2,
 		"set_code": "tower_defense",
 		"set_position": 21,
+		"text": "<b> When Revealed: </b> Corvus Glaive activates against you.",
```
Comment (lostie) [inline pack/mts_encounter.json:800]:
```suggestion
		"text": "<b>When Revealed:</b> Corvus Glaive activates against you.",
```

File: pack/mts_encounter.json [hunk 19]
```diff
@@ -771,9 +797,11 @@
...
 		"type_code": "treachery"
 	},
 	{
+		"boost_text": "Heal 2 damage from the active villain and give it a tough status card",
```
Comment (lostie) [inline pack/mts_encounter.json:804]:
```suggestion
		"boost_text": "Heal 2 damage from the active villain and give it a tough status card.",
```

File: pack/mts_encounter.json [hunk 20]
```diff
@@ -783,6 +811,7 @@
 		"quantity": 1,
 		"set_code": "tower_defense",
 		"set_position": 23,
+		"text": "<b> When Revealed: </b> Heal 2 damage from each villain. Give each villain a tough status card.",
```
Comment (lostie) [inline pack/mts_encounter.json:814]:
```suggestion
		"text": "<b>When Revealed:</b> Heal 2 damage from each villain. Give each villain a tough status card.",
```

File: pack/mts_encounter.json [hunk 21]
```diff
@@ -810,9 +840,9 @@
 		"position": 110,
 		"quantity": 2,
 		"scheme_crisis": 1,
+		"scheme_text": "Hinder 1[per_player] \n <b> When Defeated: </b> The player who defeated this cheme draws 1 card.",
```
Comment (lostie) [inline pack/mts_encounter.json:843]:
```suggestion
		"scheme_text": "Hinder 1[per_player].\n<b>When Defeated:</b> The player who defeated this scheme draws 1 card.",
```

File: pack/mts_encounter.json [hunk 22]
```diff
@@ -831,6 +861,7 @@
 		"set_code": "thanos",
 		"set_position": 1,
 		"stage": 1,
+		"text": "Stalwart. (This character cannot be stunned or confused.) \n <b> Forced Response: </b> After the [[infinity stone]] deck runs out, give Thanos 1 facedown boost card.",
```
Comment (lostie) [inline pack/mts_encounter.json:864]:
```suggestion
		"text": "Stalwart. <i>(This character cannot be stunned or confused.)</i>\n<b>Forced Response:</b> After the [[infinity stone]] deck runs out, give Thanos 1 facedown boost card.",
```

File: pack/mts_encounter.json [hunk 23]
```diff
@@ -938,6 +973,7 @@
 		"quantity": 1,
 		"set_code": "thanos",
 		"set_position": 7,
+		"text": "Attach to Thanos. \n <b> Forced Interrupt </b> When Thanos would take any amount of damage, reduce that amount by 1. /n <b> Hero Response: <b> After a hero makes a basic attack against Thanos, spend [energy] [physical] resources → discard this card.",
```
Comment (lostie) [inline pack/mts_encounter.json:976]:
```suggestion
		"text": "Attach to Thanos.\n<b>Forced Interrupt</b>: When Thanos would take any amount of damage, reduce that amount by 1.\n <b>Hero Response:<b> After a hero makes a basic attack against Thanos, spend [energy] [physical] resources → discard this card.",
```

File: pack/mts_encounter.json [hunk 24]
```diff
@@ -953,6 +989,7 @@
 		"quantity": 1,
 		"set_code": "thanos",
 		"set_position": 8,
+		"text": "Attach to Thanos. \n Thanos gains retaliate 1. <b> Hero Response: </b> After a hero makes a basica ttack against Thanos, spend [mental] [strength] resources → discard this card.",
```
Comment (lostie) [inline pack/mts_encounter.json:992]:
```suggestion
		"text": "Attach to Thanos.\nThanos gains retaliate 1.<b>Hero Response</b>: After a hero makes a basic attack against Thanos, spend [mental] [strength] resources → discard this card.",
```

File: pack/mts_encounter.json [hunk 25]
```diff
@@ -967,6 +1004,7 @@
 		"quantity": 2,
 		"set_code": "thanos",
 		"set_position": 9,
+		"text": "Attach to Thanos. [star] <b> Forced Interrupt: </b> When thano's activates, put the top card of the [[infinity stone]] deck into play. At the end of this activation, discard Master of the Stones.",
```
Comment (lostie) [inline pack/mts_encounter.json:1007]: This card is missing:

```json
"attack": 1,
```

and

```json
"scheme": 1,
```

```suggestion
		"text": "Attach to Thanos. [star] <b>Forced Interrupt:</b> When Thanos activates, put the top card of the [[infinity stone]] deck into play. At the end of this activation, discard Master of the Stones.",
```

File: pack/mts_encounter.json [hunk 26]
```diff
@@ -980,6 +1018,7 @@
 		"quantity": 2,
 		"set_code": "thanos",
 		"set_position": 11,
+		"text": "<b> When Revealed (Alter-Ego): </b> Thanos schemes. \n <b> When Revealed (Hero): </b> Thanos attacks you. That attack gains overkill and piercing.",
```
Comment (lostie) [inline pack/mts_encounter.json:1021]:
```suggestion
		"text": "<b>When Revealed (Alter-Ego)</b>: Thanos schemes.\n<b>When Revealed (Hero)</b>: Thanos attacks you. That attack gains overkill and piercing.",
```

File: pack/mts_encounter.json [hunk 27]
```diff
@@ -992,9 +1031,12 @@
 		"quantity": 2,
 		"set_code": "thanos",
 		"set_position": 13,
+		"text": "Incite 1. \n <b> When Revealed: </b> Give Thanos a tough status card. If you cannot, place 2 threat on the main scheme.",
```
Comment (lostie) [inline pack/mts_encounter.json:1034]:
```suggestion
		"text": "Incite 1.\n<b>When Revealed</b>: Give Thanos a tough status card. If you cannot, place 2 threat on the main scheme.",
```

File: pack/mts_encounter.json [hunk 28]
```diff
@@ -992,9 +1031,12 @@
...
+		"boost_text": "[star] <b>Boost</b> Give Thanos a tough status card.",
```
Comment (lostie) [inline pack/mts_encounter.json:1035]: This should be defined alphabetically, before `code`

```suggestion
		"boost_text": "Give Thanos a tough status card.",
```

File: pack/mts_encounter.json [hunk 29]
```diff
@@ -992,9 +1031,12 @@
...
 		"type_code": "treachery"
 	},
 	{
+		"boost_text": "Discard the top card of the [[infinity stone]] deck. Apply its boos icons [boost] for this activation as if it were a boost card.",
```
Comment (lostie) [inline pack/mts_encounter.json:1039]:
```suggestion
		"boost_text": "Discard the top card of the [[infinity stone]] deck. Apply its boost icons ([boost]) for this activation as if it were a boost card.",
```

File: pack/mts_encounter.json [hunk 30]
```diff
@@ -1004,6 +1046,7 @@
 		"quantity": 2,
 		"set_code": "thanos",
 		"set_position": 15,
+		"text": "<b> When Revealed: </b> Give Thanos 1 facedown boost card.",
```
Comment (lostie) [inline pack/mts_encounter.json:1049]:
```suggestion
		"text": "<b>When Revealed:</b>Give Thanos 1 facedown boost card.",
```

File: pack/mts_encounter.json [hunk 31]
```diff
@@ -1032,6 +1076,7 @@
 		"scheme_boost": 1,
 		"set_code": "thanos",
 		"set_position": 19,
+		"text": "<b> When Revealed: </b> Put the top card of the [[infinity stone]] deck into play.",
```
Comment (lostie) [inline pack/mts_encounter.json:1079]:
```suggestion
		"text": "<b>When Revealed:</b> Put the top card of the [[infinity stone]] deck into play.",
```

**Review by lostie — COMMENTED** [review]
I've reviewed Thanos cards now.

It looks much better, but there's still a lot of unnecessary spaces in the text, so it would be good for you to review it and remove them all in one go.

File: pack/mts_encounter.json (hunk 9 above)
Comment (Jensen1989) [reply pack/mts_encounter.json:536]: added in, thanks!

**Comment by Jensen1989** [comment]
I split it into smaller pull requests, I seem to have made a lot of mistakes, so smaller bits is probably better

**Pull request closed without merging**